cocoa/src/toga_cocoa/dialogs.py

```diff
@@ -46,21 +46,22 @@ def __init__(
         self.build_dialog(**kwargs)
 
         self.native.beginSheetModalForWindow(
-            interface.window._impl.native, completionHandler=completion_handler
+            interface.window._impl.native,
+            completionHandler=completion_handler,
         )
 
     def build_dialog(self):
         pass
 
     def completion_handler(self, return_value: int) -> None:
-        self.on_result(self, None)
+        self.on_result(None, None)
```

Comment:
Here and in other places: If I understand the docs correctly, the 1st argument should be the Window instance (interface layer) that the dialog belongs to. See https://toga.readthedocs.io/en/latest/reference/api/window.html#toga.window.DialogResultHandler. Or is this out of date?

Reply:
This is an area where the code is (slowly) migrating. Every time we assign a handler, we wrap it (`wrapped_handler`), which binds an instance to the handler. When the handler is eventually invoked, the [first argument (`widget`) is actually ignored](https://github.com/beeware/toga/blob/main/core/src/toga/handlers.py#L72) - and the bound instance is what is actually used. 

As we go through the audit, we've been converting this first argument in invocation to `None` to make this at least somewhat explicit; once we've finished the audit and have 100% coverage (which we're a couple of weeks away from having), we'll be able to drop the argument completely, and be certain that we've found all the places in the code that need to be updated.

Reply:
> When the handler is eventually invoked, the [first argument (widget) is actually ignored](https://github.com/beeware/toga/blob/main/core/src/toga/handlers.py#L72) - and the bound instance is what is actually used.

I do remember this now. In that case, isn't the type hint a bit confusing? For example, `info_dialog()` is supposed to take a [Optional](https://docs.python.org/3/library/typing.html#typing.Optional)[[DialogResultHandler](https://toga.readthedocs.io/en/latest/reference/api/window.html#toga.window.DialogResultHandler)[[None](https://docs.python.org/3/library/constants.html#None)]] argument, where the `DialogResultHandler` is specified as taking a window instance as a first argument. However, this is only the case for the internally wrapped handler but not the actual function that the user passes, right?

Reply:
Oh, the type hint is complete fantasy :-) However, it's what is strictly required right now to make PyCharm not have a hissy fit. Once we purge the first argument, we can clean up this Protocol definition significantly.

Reply:
Ok, thanks for the explanation!